plugins/system-update/click/manager_impl.cpp

```diff
@@ -369,8 +369,19 @@ void ManagerImpl::parseMetadata(const QJsonArray &array)
 
         auto version = object["version"].toString();
         auto icon_url = object["icon"].toString();
-        auto url = object["download"].toString();
-        auto download_sha512 = object["download_sha512"].toString();
+
+        auto downloads = object["downloads"].toArray();
+        QJsonObject download;
+
+        foreach (const auto &value, downloads) {
+            auto download_obj = value.toObject();
+            if (download_obj["channel"].toString() == Helpers::getSystemCodename())
+              download = download_obj;
```

Comment:
There's a problem here: we should consider **all** the matching packages and add them to our list. So, all the code from this point to the `model->add(update)` line below should be executed if the `if` condition is met.

I suggest to refactor the code a bit: move this foreach loop (by the way, given that we are using c++11 I'd vote for using `for (const auto &value: downloads)` instead) right after the end of the `if (dbUpdate)` condition above, and move all the rest of the loop into a separate function, that you can call from this point.

All the above holds unless we are sure that there'll always be only a single package matching our system channel; if we are sure that there'll always be only one, then you can disregard my comment (but please add a `break` after this line, and fix the indentation of the `if` block to use 4 spaces, not 2). You'll also need to add a
```
   if (download.isEmpty()) continue;
```
line after the `for` block, because it might be that the foreach loop doesn't find any package matching our system channel, and then we need to ignore them all.

Reply:
The downloads array will only have one of each channel, and will always have one that matches getSystemCodename since we query the api with channel parameter.

example 
```
"downloads": [
                    {
                        "channel": "vivid",
                        "download_url": "URL.com",
                        "download_sha512": "XXX",
                        "version": "1",
                        "revision": 1
                    },
                    {
                        "channel": "xenial",
                        "download_url": "URL.com",
                        "download_sha512": "XXX",
                        "version": "1",
                        "revision": 1
                    }
                ]
```

Also the reason i used foreach is that just some lines bellow foreach was used so i want to keep it consistent across the code

Reply:
Mmm... AFAIU, it's possible for a package to have a different version and revision for the various channel (for example, the developer might stop making releases for vivid).

I think you should read the version and revision from the download object; that probably means that you need to move the code that you added to the beginning of the loop, where these values are first obtained to check the update database.

Reply:
You are correct, that does make sense, since the other value seems to only reflect the latest revision and not on the channels. sorry i missed that one. Made a new commit